drivers/can/can_mcan.c

```diff
@@ -797,12 +797,24 @@ void can_mcan_line_1_isr(const struct device *dev)
 	uint32_t ir;
 	int err;
 
-	err = can_mcan_read_reg(dev, CAN_MCAN_IR, &ir);
-	if (err != 0) {
-		return;
-	}
+	do {
+		err = can_mcan_read_reg(dev, CAN_MCAN_IR, &ir);
```

Comment:
Maybe I'm missing something, but I don't see how this reduces the number of calls to `can_mcan_read_reg`. It is still called at least once and afterwards as long as the condition is not met.

However, with the `do { ... } while (true)` loop it's not completely obvious anymore what the stop condition is. I think I prefer keeping the existing implementation.

Reply:
Thank you for the fast response. Basically my change avoids two IR register reads in case while-loop will be approached. 
Thus you got it right. 
I agree - this change is mainly cosmetic, but shall still saves few machine-cycles, which may be useful on very low budget MCUs with low speed.
I would suggest also Henrik to vote on this change. And of course I will remove it, if you both vote against. I am fine with it.

Reply:
I must admit that I too see no real issue with the current implementation and find it more readable (but I may be biased).

Reply:
I too think the current implementation is more straight forward because there is a single break condition.

Reply:
> I too think the current implementation is more straight forward because there is a single break condition.
 
Hello Alexander and thank you for your review. Well, in my implementation there is also only one break-condition, cause `while(true)` is not a break-condition ;) 
I see, majority voting against while-loop redesign. I will revert this changes and leave only debug priontouts.